tensorflow/python/ops/data_flow_ops.py

```diff
@@ -1919,16 +1961,19 @@ def peek(self, key, name=None):
     if name is None:
       name = "%s_pop" % self._name
 
+    indices, dtypes = self._get_indices_and_dtypes(indices)
+
     with ops.colocate_with(self._coloc_op):
       result = self._peek_fn(key, shared_name=self._name,
-                      dtypes=self._dtypes,
+                      indices=indices,
+                      dtypes=dtypes,
                       name=name,
                       capacity=self._capacity,
                       memory_limit=self._memory_limit)
 
-    return self._get_return_value(result)
+    return self._get_return_value(result, indices)
 
-  def get(self, key=None, name=None):
+  def get(self, key=None, indices=None, name=None):
```

Comment:
I think it's fine in this case since I don't think there are likely any users of MapStagingArea just yet (and definitely none internally), but generally adding new optional parameters should go at the end of the list, otherwise you break all callers that haven't been using named parameters.